### PR TITLE
Add format_to_n overload that accepts FMT_COMPILE

### LINKED ISSUE #1764: fmt::format_to_n does not work with FMT_COMPILE-d format string
Example code: https://godbolt.org/z/erD-_a

In v6 `fmt::format_to_n()` seems to work with `fmt::compile()`-d format string. 
In v7 though, just `fmt::format_to()`  is working with `FMT_COMPILE()`.

### PR DESCRIPTION
**Pull request by Kurkin**
Resolves #1764 

I agree that my contributions are licensed under the {fmt} license, and agree to future changes to the licensing.

### CODE REVIEW

File: include/fmt/compile.h [hunk 1]
```diff
@@ -598,6 +598,17 @@ OutputIt format_to(OutputIt out, const CompiledFormat& cf,
...
 }
+
+template <typename OutputIt, typename CompiledFormat, typename... Args,
+          FMT_ENABLE_IF(detail::is_compiled_format<CompiledFormat>::value)>
+format_to_n_result<OutputIt> format_to_n(OutputIt out, size_t n,
+                                         const CompiledFormat& cf,
+                                         const Args&... args) {
+  auto it =
+      format_to(detail::truncating_iterator<OutputIt>(out, n), cf, args...);
+  return {it.base(), it.count()};
```
Comment (vitaut) [inline include/fmt/compile.h:609]: Please merge this part into the overload below and remove this overload.

Comment (Kurkin) [reply include/fmt/compile.h]: Done

**Comment by vitaut** [comment]
Looks good but could you also add a unit test similar to https://github.com/fmtlib/fmt/blob/bd903f96ac93cccca39e0921c046c02258a3198c/test/compile-test.cc#L156

**Comment by vitaut** [comment]
Closing due to inactivity but feel free to resubmit with a test.

**Pull request closed without merging**